benches/benches/bevy_ecs/change_detection.rs

```diff
@@ -0,0 +1,102 @@
+use bevy_ecs::{
+    component::Component,
+    entity::Entity,
+    prelude::{Added, Changed},
+    world::World,
+};
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use rand::{prelude::SliceRandom, SeedableRng};
+use rand_chacha::ChaCha8Rng;
+
+criterion_group!(benches, added_detection, changed_detection,);
+criterion_main!(benches);
+
+#[derive(Component, Default)]
+#[component(storage = "Table")]
+struct Table(f32);
+#[derive(Component, Default)]
+#[component(storage = "SparseSet")]
+struct Sparse(f32);
```

Comment:
Could these be zero sized? I suppose it doesn't matter, since they're never queried anyway?